### PR TITLE
Added Random extension method

### PR DESCRIPTION
**Pull request by Amitsim**
An extension method called GetRandomElement for IEnumerable<T>. It allows for easy retrieval of a random element from any collection that implements IEnumerable<T>, including IList<T> and ICollection<T>.

Changes:
Added the Random extension method to the EnumerableExtensions class.
The method generates a random index using UnityEngine.Random.Range and returns the element at that index from the collection.
Checks are added to handle cases where the collection is null or empty, throwing appropriate exceptions (ArgumentNullException and InvalidOperationException).

Tested the method with a variety of collections, including List<T>, HashSet<T>, and arrays.

### CODE REVIEW

File: UnityUtils/Scripts/Extensions/EnumerableExtensions.cs [hunk 1]
```diff
@@ -14,5 +14,26 @@ public static void ForEach<T>(this IEnumerable<T> sequence, Action<T> action) {
...
         }
+
+
+        /// <summary>
+        /// Returns a random element from the sequence.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the sequence.</typeparam>
+        /// <param name="sequence">The sequence to select the random element from.</param>
+        /// <returns>A random element from the sequence.</returns>
+        public static T Random<T>(this IEnumerable<T> sequence)
```
Comment (adammyhre) [inline UnityUtils/Scripts/Extensions/EnumerableExtensions.cs:29]: Consider the following changes:

1. Avoid unnecessary allocation of a new list when the sequence is already an IList<T>
2. Add fallback mechanism using reservoir sampling for sequences that are not IList<T>, such as streams or lazy-loaded sequences, ie:
```csharp
        public static T Random<T>(this IEnumerable<T> sequence) {
            if (sequence == null)
                throw new ArgumentNullException(nameof(sequence));

            if (sequence is IList<T> list) {
                if (list.Count == 0)
                    throw new InvalidOperationException("Cannot get a random element from an empty collection.");
                return list[UnityEngine.Random.Range(0, list.Count)];
            }

            // Use reservoir sampling when the input is not an IList<T> ie: a stream or lazy sequence
            using var enumerator = sequence.GetEnumerator();
            if (!enumerator.MoveNext())
                throw new InvalidOperationException("Cannot get a random element from an empty collection.");

            T result = enumerator.Current;
            int count = 1;
            while (enumerator.MoveNext()) {
                if (UnityEngine.Random.Range(0, ++count) == 0) {
                    result = enumerator.Current;
                }
            }
            return result;
        }
```

**Review by Amitsim — COMMENTED** [review]
This is my first Pull Request, and I'm new to GitHub. I changed the code exactly as you suggested and tested it in Unity. I also enabled 'Allow edits by maintainers'.

File: UnityUtils/Scripts/Extensions/EnumerableExtensions.cs [hunk 2]
```diff
@@ -2,38 +2,49 @@
...
-        /// </summary>
-        /// <typeparam name="T">The type of elements in the sequence.</typeparam>
-        /// <param name="sequence">The sequence to iterate over.</param>
-        /// <param name="action">The action to perform on each element.</param>    
-        public static void ForEach<T>(this IEnumerable<T> sequence, Action<T> action) {
-            foreach (var item in sequence) {
-                action(item);
-            }
-        }
+	public static class EnumerableExtensions {
```
Comment (adammyhre) [inline UnityUtils/Scripts/Extensions/EnumerableExtensions.cs]: nit. clean up spacing, it seems to have changed to 8 chars instead of 4

**Review by Amitsim — COMMENTED** [review]
I changed from 8 char to 4 char.

**Comment by adammyhre** [comment]
Not bad, but for some reason now all opening curly brackets are on new lines.  I will fix it.

**Pull request merged**